allensdk/brain_observatory/behavior/session_apis/data_io/behavior_nwb_api.py

```diff
@@ -0,0 +1,195 @@
+import datetime
+import uuid
+
+import numpy as np
+import pandas as pd
+import pytz
+
+from pynwb import NWBHDF5IO, NWBFile
+
+import allensdk.brain_observatory.nwb as nwb
+from allensdk.brain_observatory.behavior.metadata_processing import (
+    get_expt_description
+)
+from allensdk.brain_observatory.behavior.session_apis.abcs import (
+    BehaviorBase
+)
+from allensdk.brain_observatory.behavior.schemas import (
+    BehaviorTaskParametersSchema, OphysBehaviorMetadataSchema)
+from allensdk.brain_observatory.behavior.trials_processing import (
+    TRIAL_COLUMN_DESCRIPTION_DICT
+)
+from allensdk.brain_observatory.nwb.metadata import load_pynwb_extension
+from allensdk.brain_observatory.nwb.nwb_api import NwbApi
+from allensdk.brain_observatory.nwb.nwb_utils import set_omitted_stop_time
+
+load_pynwb_extension(OphysBehaviorMetadataSchema, 'ndx-aibs-behavior-ophys')
+load_pynwb_extension(BehaviorTaskParametersSchema, 'ndx-aibs-behavior-ophys')
+
+
+class BehaviorNwbApi(NwbApi, BehaviorBase):
+    """A data fetching class that serves as an API for fetching 'raw'
+    data from an NWB file that is both necessary and sufficient for filling
+    a 'BehaviorOphysSession'.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.filter_invalid_rois = kwargs.pop("filter_invalid_rois", False)
+        super(BehaviorNwbApi, self).__init__(*args, **kwargs)
+
+    def save(self, session_object):
```

Comment:
`NwnApi` should define an abstract method `save` which is overidden by subclasses